### ISSUE TITLE
feature request: ability to disable specific checkers

### ISSUE THREAD

**Issue by scop** [Antonboom/testifylint#12]
Would be nice to be able to disable specific checkers.

Use cases:
- Enable all, but disable some specific checker, `--enable-all --disable=some-specific-checker`
- Use with golangci-lint and use its defaults for this linter otherwise, but disable some specific one instead of having to repeat the whole config
    - Workaround:
        ```yaml
        issues:
          exclude:
            - "^some-specific-checker: "
        ```

**Comment by Antonboom** [comment]
Hi, thank you for report!

Initially it was exactly so, but I was worried about the thoughtless enabling of linters (as happens in golangci-lint). I think I'll return a more detailed configuration back 👌

For information, which checkers are you disabled?
For example, I'm currently working on minimizing `require-error` false positives.

**Comment by scop** [comment]
Hi, thanks for considering this!

`require-error` is the one I'm evaluating and considering to disable. It is certainly useful, but somewhat prone to false positives in the version included with golangci-lint 1.55.0. Great to hear you're working on addressing some of them :+1: 

I don't have very well thought out or organized feedback at the moment, nor am I declaring these as definite false positives nor good programming practices necessarily, but here are a couple of cases off the top of my tongue I've looked at today:

1) If an assertion is the last one run in a test, it doesn't matter if it's an assert or a require. An assert might be desirable in case one wants to later add some more assertions to the test that are not dependent on this one. Would prefer if required-error would not be issued for these cases.

2) In table driven tests, it's not uncommon to have expected outcomes and errors. When we know that _some_ error has occurred, we may want to assert something on the outcome (e.g. that it's nil or empty, which might what we have as the expected outcome in the table as well) even if the error was not the expected one. So e.g.
    ```go
    outcome, err := something()
    if test.expectedError == nil {
        require.NoError(t, err)
        // some tests on expected outcome
    } else {
        assert.ErrorIs(t, err, test.expectedError)  // <-- if this gets turned into a require, the following emptiness check never runs
        assert.Empty(t, outcome)
    }
    ```
   I can imagine this would likely be a tough nut to crack in the linter.

3) Guarding NoError asserts and only continuing with further tests on something with an `if` is fine, e.g.
    ```go
    if assert.NoError(t, err) {
        // further things here that depend on `err` not occurred
    }
    // some more code and assertions that are *not* dependent on whether the above `err` occurred may follow here
    ```
    or
    ```go
    if b, err := json.Marshal(foo); assert.NoError(t, err) {
        // further things here that depend on `err` not occurred
    }
    // some more code and assertions that are *not* dependent on whether the above `err` occurred may follow here
    ```
    If these are turned into `require`, the "some more code and assertions" never run, even though they are independent. One might argue that these are not well constructed tests and should be split into sub ones or something, but :shrug:
